pkg/virt-controller/watch/migration.go

```diff
@@ -140,15 +140,15 @@ func StartMigrationTargetPod(v services.VMService, migration *v1.Migration) erro
 
 	//TODO:  detect collisions
 	conflicts := []string{}
-	for k, v := range migration.Spec.DestinationNodeSelector {
+	for k, v := range migration.Spec.NodeSelector {
```

Comment:
If the old code had an explicit hostname node selection it should be dropped, not an error.
probably prepend  
delete(migration.Spec.DestinationNodeSelector, 'kubernetes.io/hostname')
and ignore an error that means a missing key.